source/agora/consensus/ValidatorSet.d

```diff
@@ -319,6 +319,33 @@ public class ValidatorSet
         }
     }
 
+    /***************************************************************************
+
+        Gets the number of valid validators at the block height.
+
+        Params:
+            block_height = the height of proposed block
+
+    ***************************************************************************/
+
+    public ulong getValidatorCount (ulong block_height) @safe nothrow
+    {
+        ulong validator_count = 0;
+        try
+        {
+            ulong height = (block_height >= Enrollment.ValidatorCycle) ?
+                block_height - Enrollment.ValidatorCycle + 1 : 0;
+            () @trusted {
+                validator_count = this.db.execute("SELECT count(*) FROM validator_set WHERE enrolled_height >= ?", height).oneValue!ulong;
```

Comment:
I think the code below is right because the `height` value that you have calculated is actually the `enrolled_height` plus 1.

```
validator_count = this.db.execute("SELECT count(*) FROM validator_set WHERE enrolled_height >= ?", height-1).oneValue!ulong;
```

Reply:
I thought after watching ClearExpiredValidators.
Please explain in detail what the problem is.
ClearExpiredValidators :
enrolled_height <= block_height - Enrollment.ValidatorCycle
getValidatorCount:
enrolled_height >= block_height - Enrollment.ValidatorCycle + 1